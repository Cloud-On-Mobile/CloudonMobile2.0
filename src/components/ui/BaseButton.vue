<script lang="ts" setup>
import { defineProps } from 'vue'
interface Props {
  theme: string
}
const { theme = 'active' } = defineProps<Props>()
</script>

<template>
  <button :class="`base-button base-button--${theme}`">
    <slot />
  </button>
</template>

<style lang="scss" scoped>
.base-button {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 16px 24px;
  width: 100%;
  height: 56px;
  border-radius: 12px;
  margin-top: 56px;
  color: $color-text-buttons;

  &--active {
    background-color: $color-background-special;
  }

  &--inactive {
    background-color: $color-background-inactive;
  }
}
</style>
