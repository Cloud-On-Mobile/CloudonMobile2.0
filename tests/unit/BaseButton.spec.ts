import { shallowMount } from '@vue/test-utils'
import BaseButton from '@/components/ui/BaseButton.vue'

describe('BaseButton', () => {
  it('renders with active theme when passed as prop', () => {
    const theme = 'active'
    const wrapper = shallowMount(BaseButton, { props: { theme }})
    expect(wrapper.attributes('class')).toBe('base-button base-button--active')
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders with inactive theme when passed as prop', () => {
    const theme = 'inactive'
    const wrapper = shallowMount(BaseButton, { props: { theme }})
    expect(wrapper.attributes('class')).toBe('base-button base-button--inactive')
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders with new-file theme when passed as prop', () => {
    const theme = 'new-file'
    const wrapper = shallowMount(BaseButton, { props: { theme }})
    expect(wrapper.attributes('class')).toBe('base-button base-button--new-file')
  })
})
