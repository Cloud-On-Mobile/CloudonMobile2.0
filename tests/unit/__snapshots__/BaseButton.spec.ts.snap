// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton renders with active theme when passed as prop 1`] = `
<button
  class="base-button base-button--active"
>
  
  
</button>
`;

exports[`BaseButton renders with inactive theme when passed as prop 1`] = `
<button
  class="base-button base-button--inactive"
>
  
  
</button>
`;
